api/submission/fetch_course_report.go

```diff
@@ -0,0 +1,26 @@
+package submission
+
+import (
+    "github.com/edulinq/autograder/api/core"
+    "github.com/edulinq/autograder/report"
+)
+
+type FetchCourseReportRequest struct {
+    core.APIRequestCourseUserContext
+    core.MinRoleGrader
+}
+
+type FetchCourseReportResponse struct {
+    CourseReport *report.CourseScoringReport `json:"course-report"`
+}
+
+func HandleFetchCourseReport(request *FetchCourseReportRequest) (*FetchCourseReportResponse, *core.APIError) {
+    gettingCourseReport, err := report.GetCourseScoringReport(request.Course);
```

Comment:
I don't like this variable name.
The "getting" pattern does not really make sense for anything that is not a promise.
I would say the obvious name here should be "courseReport".